arrayqueues/portable_queue.py

```diff
@@ -0,0 +1,69 @@
+"""The code below has been entirely taken from the following GitHub gist:
+https://gist.github.com/FanchenBao/d8577599c46eab1238a81857bb7277c9
+"""
+
+import multiprocessing
+
+
+class SharedCounter(object):
+    """ A synchronized shared counter.
+
+    The locking done by multiprocessing.Value ensures that only a single
+    process or thread may read or write the in-memory ctypes object. However,
+    in order to do n += 1, Python performs a read followed by a write, so a
+    second process may read the old value before the new one is written by the
+    first process. The solution is to use a multiprocessing.Lock to guarantee
+    the atomicity of the modifications to Value.
+
+    This class comes almost entirely from Eli Bendersky's blog:
+    http://eli.thegreenplace.net/2012/01/04/shared-counter-with-pythons-multiprocessing/
+
+    """
+
+    def __init__(self, n=0):
+        self.count = multiprocessing.Value('i', n)
+
+    def increment(self, n = 1):
+        """ Increment the counter by n (default = 1) """
+        with self.count.get_lock():
+            self.count.value += n
+
+    @property
+    def value(self):
+        """ Return the value of the counter """
+        return self.count.value
+
+
+class Queue(multiprocessing.queues.Queue):
```

Comment:
I think it is just under multiprocessing

Reply:
it looks like the importing ways from multiprocessing can do funny things to the multiprocessing context, that's why I kept it that way